components/application-connectivity-validator/internal/controller/cachesynccontroller.go

```diff
@@ -0,0 +1,220 @@
+
+	"github.com/kyma-project/kyma/components/application-operator/pkg/apis/applicationconnector/v1alpha1"
+	clientset "github.com/kyma-project/kyma/components/application-operator/pkg/client/clientset/versioned"
+	log "github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/errors"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	informers "github.com/kyma-project/kyma/components/application-operator/pkg/client/informers/externalversions/applicationconnector/v1alpha1"
+	listers "github.com/kyma-project/kyma/components/application-operator/pkg/client/listers/applicationconnector/v1alpha1"
+)
+
+const (
+	// SuccessSynced is used as part of the Event 'reason' when a Application is synced
+	SuccessSynced = "Synced"
+	// ErrResourceExists is used as part of the Event 'reason' when a Application fails
+	// to sync due to a Deployment of the same name already existing.
+	ErrResourceExists = "ErrResourceExists"
+
+	// MessageResourceExists is the message used for Events when a resource
+	// fails to sync due to a Deployment already existing
+	MessageResourceExists = "Resource %q already exists and is not managed by Application"
+	// MessageResourceSynced is the message used for an Event fired when a Application
+	// is synced successfully
+	MessageResourceSynced = "Application resource synced successfully"
+)
+
+// Controller is the controller implementation for Application resources
+type Controller struct {
+	// clientset is a clientset for our own API group
+	clientset clientset.Interface
+
+	applicationLister listers.ApplicationLister
+	applicationSynced cache.InformerSynced
+
+	// workqueue is a rate limited work queue. This is used to queue work to be
+	// processed instead of performing it as soon as a change happens. This
+	// means we can ensure we only process a fixed amount of resources at a
+	// time, and makes it easy to ensure we are never processing the same item
+	// simultaneously in two different workers.
+	workqueue workqueue.RateLimitingInterface
+
+	// Name of the application CR the controller syncs the cache for
+	appName string
+
+	// Cache to store Application CRs
+	appCache *gocache.Cache
+}
+
+func NewController(
+	clientset clientset.Interface,
+	applicationInformer informers.ApplicationInformer,
+	appName string,
+	appCache *gocache.Cache) *Controller {
+
+	controller := &Controller{
+		clientset:         clientset,
+		applicationLister: applicationInformer.Lister(),
+		applicationSynced: applicationInformer.Informer().HasSynced,
+		workqueue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "Applications"),
+		appName:           appName,
+		appCache:          appCache,
+	}
+
+	applicationInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: controller.enqueueApplication,
+			UpdateFunc: func(old, new interface{}) {
+				controller.enqueueApplication((new))
+			},
+			DeleteFunc: controller.enqueueApplication,
```

Comment:
When application is deleted, the application-connectivity-validator as its part will we deleted as well. Maybe we don't need to react to this event since the pod itself will be deleted.

Reply:
Fixed